python/ray/data/_internal/logical/operators/all_to_all_operator.py

```diff
@@ -115,3 +115,18 @@ def __init__(
         )
         self._key = key
         self._aggs = aggs
+
+
+class Limit(AbstractAllToAll):
```

Comment:
Should this extend PhysicalOperator instead of AbstractAllToAll?

Reply:
This is the logical operator. Not used right now, but I added this for completeness of the `Datastream.limit` API.
I think limit is also a kind of all-to-all logical operator. So this should make sense? Or do you prefer to have it directly extend to `LogicalOperator`.

Reply:
Don't have a strong opinion, but it makes sense that it should extend LogicalOperator to me and not AllToAll, since there's technically not any all:all data dependency between rows.

Reply:
Makes sense. Fixed.